### ISSUE TITLE
Rule Change: Update execScript message in no-implied-eval rule

### ISSUE THREAD

**Issue by TKDev7** [eslint/eslint#19927]
### What rule do you want to change?

no-implied-eval

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
execScript('wibble');
```

### What does the rule currently do for this code?

Flags it with:

```
Implied eval. Consider passing a function instead of a string.
```

### What will the rule do after it's changed?

No longer warn about execScript() calls. 

In 2025, usage of IE's global execScript() is an extreme edge case. Developers shouldn’t expect the execScript name to be restricted due to a deprecated function in an obsolete browser.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

- `execScript` was an Internet Explorer-specific function (now obsolete)
- The warning message is misleading since `execScript` doesn't accept functions
- Developers needing to block `execScript` can use `no-restricted-globals` instead

**Comment by nzakas** [comment]
Thanks for the suggestion. I don't see the reason to remove checking for `execScript()`. The rule already checks to see if `execScript()` is a global variable before flagging it, which should protect against false positives for other functions called `execScript()`.

We could potentially update the error message to be less confusing. I don't think I'd want to do anything more than that.

**Comment by TKDev7** [comment]
I didn't consider #19923 to be a bug because the `no-eval` rule behaves similarly by flagging all `eval`, including non-global ones. The documentation justifies this behavior to prevent false negatives in cases like:

```js
module.exports = function(eval) {
    // If the value of this `eval` is built-in `eval` function, this is a
    // call of direct `eval`.
    eval("const a = 0");
};
```

The same logic likely applies to `no-implied-eval`.

**Comment by nzakas** [comment]
Sorry, I'm not sure how to interpret your comment. Are you agreeing with me? Disagreeing? What do you think should be done here?

**Comment by TKDev7** [comment]
Sorry, to clarify—I don’t think #19923 is a false positive for the reason I mentioned earlier. However, since the team disagrees, the only actionable step here is to update the execScript error message.

**Comment by nzakas** [comment]
Got it. I agree, let's update the error message.
